cpp/basix.cpp

```diff
@@ -117,22 +179,28 @@ const char* basix::mapping_name(int handle)
   return mapping::type_to_str(_registry[handle]->mapping_type()).c_str();
 }
 
-Eigen::ArrayXXd basix::geometry(const char* cell_type)
+int basix::cell_geometry_num_points(const char* cell_type)
 {
   cell::type ct = cell::str_to_type(cell_type);
-  return cell::geometry(ct);
+  return cell::geometry(ct).rows();
 }
 
-std::vector<std::vector<std::vector<int>>>
-basix::topology(const char* cell_type)
+int basix::cell_geometry_dimension(const char* cell_type)
 {
   cell::type ct = cell::str_to_type(cell_type);
-  return cell::topology(ct);
+  return cell::geometry(ct).cols();
+}
+
+void basix::cell_geometry(const char* cell_type, double* points)
```

Comment:
Does this function get called frequently? (had a dynamic memory allocation in it)

Reply:
No, this was changed to get rid of Eigen. I believe dolfinx will call this once then store the points